mod/groups/start.php

```diff
@@ -680,41 +695,6 @@ function groups_get_invited_groups($user_guid, $return_guids = false, $options =
 }
 
 /**
- * Join a user to a group, add river event, clean-up invitations
- *
- * @param ElggGroup $group
- * @param ElggUser  $user
- * @return bool
- */
-function groups_join_group($group, $user) {
-
-	// access ignore so user can be added to access collection of invisible group
-	$ia = elgg_set_ignore_access(TRUE);
```

Comment:
This shouldn't be there to start with. I looked through the stack and there is no reason for ignored access. If handlers need it, they can add it themselves. `$group` is already loaded and access hooks don't check access.